app/src/main/java/com/example/myapplication/SignUpDetailsActivity.kt

```diff
@@ -21,24 +25,65 @@ class SignUpDetailsActivity : AppCompatActivity() {
         val nextButton = findViewById<Button>(R.id.btnNext)
 
         nextButton.setOnClickListener {
-            val fullName = fullNameEditText.text.toString()
-            val validEmail = validEmailEditText.text.toString()
-            val phoneNumber = phoneNumberEditText.text.toString()
-            val strongPassword = strongPasswordEditText.text.toString()
+            val fullName = fullNameEditText.text.toString().trim()
+            val email = validEmailEditText.text.toString().trim()
+            val phone = phoneNumberEditText.text.toString().trim()
+            val password = strongPasswordEditText.text.toString().trim()
             val isTermsAccepted = termsAndConditionsCheckBox.isChecked
 
-            if (isTermsAccepted) {
-                Toast.makeText(this, "You have successfully registered!", Toast.LENGTH_SHORT).show()
-            } else {
-                Toast.makeText(this, "Please accept the Terms and Conditions", Toast.LENGTH_SHORT)
-                    .show()
+            when {
+                !credentialsManager.isValidFullName(fullName) -> {
+                    fullNameEditText.error = "Full Name is required"
+                }
+
+                !credentialsManager.isValidEmail(email) -> {
+                    validEmailEditText.error = "Invalid email format"
+                }
+
+                !credentialsManager.isValidPhoneNumber(phone) -> {
+                    phoneNumberEditText.error = "Phone number must be at least 10 digits"
+                }
+
+                !credentialsManager.isValidPassword(password) -> {
+                    strongPasswordEditText.error = "Password must be at least 8 characters"
+                }
+
+                !credentialsManager.isTermsAccepted(isTermsAccepted) -> {
+                    Toast.makeText(
+                        this,
+                        "Please accept the Terms and Conditions",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                }
+
+                else -> {
+                    if (credentialsManager.ValidateCredentialsForSignUp(
+                            fullName,
+                            email,
+                            phone,
+                            password,
+                            isTermsAccepted
+                        )
+                    ) {
+                        Toast.makeText(
```

Comment:
You have an if with both bodies doing 'exactly' the same with difference of displayed string
This is a good moment to export it to a method to avoid duplication